pkg/utils/customresource/typegenerics.go

```diff
@@ -0,0 +1,49 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package customresource
+
+import (
+	"fmt"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+)
+
+// RetrieveLatestUpdateTime takes an arbitrary custom resource and retrives its latest update time
+// based on the resource's metadata
+func RetrieveLatestUpdateTime(customResource interface{}) (*time.Time, error) {
+	metaAccessor, maAssertable := customResource.(metav1.ObjectMetaAccessor)
+	resourceKR, krAssertable := customResource.(duckv1.KRShaped)
+	if !maAssertable || !krAssertable {
+		return nil, fmt.Errorf("The specified customResource does not support an expected type assertion (ObjectMetaAccessor: %v, KRShaped: %v) %#v", maAssertable, maAssertable, customResource)
+	}
```

Comment:
It seems we can simply this logic. In the `controller.go` where we setup the informer, we already attempt to convert the interface to `*brokerv1beta1.Broker` or `*brokerv1beta1.Trigger`. And they are for sure matching these types.

Reply:
Instead of hardcoding towards trigger and broker (which I indeed had in the first version), I made this a generic util, which can be re-used on other custom resources. The error handling above is just meant to protect against inappropriate usage e.g. when some other resource or a possibly incompatible future version of it is passed -- seems like a good practice for a util/helper/library kind of methods.

Reply:
I'm being nit picking here. I didn't mean that this should _only_ accept broker and trigger. But instead of interface, probably something more deterministic, e.g. `metav1.ObjectMetaAccessor`. Then you don't have to do the cast check in this func.

Reply:
Ah, I see the point, have simplified this in the latest update. The methods come from two different interfaces, so we have one remaining casting involved.